### PR TITLE
Create disable-user action

### PR DESCRIPTION
**Pull request by whaught**
Issue https://github.com/google/exposure-notifications-verification-server/issues/975

## Proposed Changes

* API to disable a user's login

Next PR will integrate this into a UI for System admin

**Release Note**

```release-note
Button for system admins to disable login for a user
```

### CODE REVIEW

File: cmd/server/assets/users/show.html [hunk 1]
```diff
@@ -37,15 +38,67 @@ <h1>{{$user.Name}}</h1>
...
+          {{else}}
+          <div class="card-text mb-3">Disabled</div>
+          {{end}}
         </div>
 
         <a href="/users/{{$user.ID}}/reset-password" data-method="POST" class="btn btn-primary btn-block">Send password reset</a>
       </div>
     </div>
 
+    {{if $currentUser.Admin}}
```
Comment (sethvargo) [inline cmd/server/assets/users/show.html:59]: I'm -1 on including admin-related actions outside of the admin console. I think we should only have these pages for realm admins and system admin operations need to take place on the system admin console.

File: internal/auth/firebase.go [hunk 2]
```diff
@@ -163,6 +163,23 @@ func (f *firebaseAuth) CreateUser(ctx context.Context, name, email, pass string,
 	return true, nil
 }
 
+// DisableUser prevents a user from logging in by disabling their account.
+func (f *firebaseAuth) DisableUser(ctx context.Context, email string) error {
```
Comment (sethvargo) [inline internal/auth/firebase.go:167]: Why do we need to disable the user in firebase versus doing a local disable in the database?

File: internal/routes/server.go [hunk 3]
```diff
@@ -283,6 +283,13 @@ func Server(
 		userSub.Handle("/{id}", userController.HandleUpdate()).Methods("PATCH")
 		userSub.Handle("/{id}", userController.HandleDelete()).Methods("DELETE")
 		userSub.Handle("/{id}/reset-password", userController.HandleResetPassword()).Methods("POST")
+
+		userAdminSub := r.PathPrefix("/users").Subrouter()
```
Comment (sethvargo) [inline internal/routes/server.go:287]: I think this should be a system admin-only route.

**Review by sethvargo — CHANGES_REQUESTED** [review]

File: internal/auth/firebase.go (hunk 2 above)
Comment (whaught) [reply internal/auth/firebase.go:167]: if there were an abusive user this stops all access, it's also flippable so you could disable and then enable without having to re-join all their realms

Comment (sethvargo) [reply internal/auth/firebase.go:167]: We can set `DeletedAt` on the user in the database and achieve the same effect (without removing them from all the realms). I'm worried about a situation where the developer's firebase is attached to multiple systems. I dont think they'd expect disabling a user here to disable them in other systems.

**Comment by whaught** [comment]
/close

I'll revisit pieces of this, but going forward with just delete instead of disable

**Pull request closed without merging**